README.md

```diff
@@ -11,78 +11,83 @@ related-issues: (will contain links to implementation PRs)
+  - `nix edit -f . package-attr` works, though that's not yet stable (it relies on the `nix-command` feature being enabled) and doesn't work with packages that don't set `meta.position` correctly).
 - `all-packages.nix` frequently causes merge conflicts. It's a point of contention for all new packages
 
 # Detailed design
 [design]: #detailed-design
 
-All attributes at the root of nixpkgs (`pkgs.<name>`) whose definition is of the form
-
-```nix
-{
-  <name> = pkgs.callPackage ../some/dir { };
-}
-```
-
-will be become eligible to be transformed as follows:
-- Move `pkgs/some/dir` to `pkgs/auto/<name>`
-- Fix any references to that folder (e.g. in update scripts) 
-- Remove the original definition
-
-These attributes will newly be defined by listing all directories in `pkgs/auto` using `builtins.readDir` and calling `pkgs.callPackage` on all of them, which then gets added to the `pkgs` scope.
-
-Attributes whose definition aren't of the above form won't be changed, so e.g. the following definition in `all-packages.nix` won't be changed:
-```nix
-{
-  syncplay-nogui = syncplay.override { enableGUI = false; };
-}
-```
-
-However, if such definitions can be refactored into the above form they will become eligible for the transformation.
-
-## Backwards compatibility symlinks
-[symlinks]: #backwards-compatibility-symlinks
+This RFC establishes the convention of `pkgs/unit/${substring 0 4 name}/${name}` "unit" directories for the definitions of the Nix packages `pkgs.${name}` in nixpkgs.
+The `pkg-fun.nix` files in all unit directories are automatically discovered, called using `pkgs.callPackage` and added to the `pkgs` set.
 
-When moving `pkgs/some/dir/default.nix` to the new `pkgs/unit/<name>/pkg-fun.nix`, a symlink will be created pointing from `pkgs/unit/some/dir/default.nix` to `pkgs/unit/<name>/pkg-fun.nix`. Reasoning:
-- Current community discussions referencing old files from the `master` branch are still valid for some time. While GitHub doesn't provide an easy way to navigate to a symlink, seeing the path to where the file has moved is better than getting an error.
-- It provides an opportunity for code referencing old paths to be updated. While it's not possible to give a deprecation warning with symlinks, users will at least be able to read it in the NixOS release notes. This doesn't occur often in practice.
+These requirements will be checked using CI:
+1. The `pkgs/unit` directory must only contain unit directories, and only in subdirectories of the form `${substring 0 4 name}/${name}`.
+2. <a id="req-ref"/> Files outside a unit directory must not reference files inside that unit directory, and the other way around.
+3. Each unit directory must have a `pkg-fun.nix` file such that `pkgs.callPackage ./pkg-fun.nix {}` evaluates to a derivation.
+4. Packages defined by unit directories must not be defined or overridden anywhere else, such as in `pkgs/top-level/all-packages.nix`.
```

Comment:
```suggestion
4. The definition of a package in the unit directory is the one `pkgs.<name>` points to.
5. To avoid problems with merges, if a package attribute is defined by a unit directory, it must not be defined in `pkgs/top-level/all-packages.nix` or `pkgs/top-level/aliases.nix`.
```